### ISSUE TITLE
Unsupported `constexpr std::bit_cast` for `std::complex`

### ISSUE THREAD

**Issue by Svalorzen** [llvm/llvm-project#94620]
Hello, I am trying to bit_cast a structure containing `std::complex`, and clang is reporting that this is unsupported. I'm attaching here an MVE:
```
#include <bit>
#include <iostream>
#include <complex>
#include <array>

using C = std::complex<double>;

int main() {
    constexpr unsigned char data[sizeof(C)] = {0};

    constexpr C c = std::bit_cast<C>(data);
    std::cout << c << '\n';

    return 0;
}
```
This seems to fail on trunk as per godbolt (https://godbolt.org/z/q8fKE7r18).

There exists a similar issue, #54018, with PR #74775, which involves bit-fields. I am not sure whether this is related at low level, but I think enabling `std::bit_cast` on a "native" standard library type is quite important. I am opening this issue as I have not been able to find any status reports on this problem.

**Comment by Svalorzen** [comment]
That's encouraging. Any chance it would work with libstdc++ as well?

**Comment by Svalorzen** [comment]
I am on Linux, and I confirm that I can compile this just fine with gcc. The only reason I discovered this problem is because I use YCM with vim, which uses LLVM in order to provide diagnostics in editor. So I started getting this error reported, then I tried to compile using clang (which on Linux I think uses stdlibc++ by default) and I confirmed the problem.

Installing libc++ on Linux is not necessarily easy so that's why I was asking whether it would be possible for clang to make this work with libstdc++ as well.

**Comment by tbaederr** [comment]
This is simply a problem with clang's implementation of `__builtin_bit_cast`.

Much simpler reproducer:
```c++
constexpr _Complex short A = {1, 2};
constexpr int B = __builtin_bit_cast(int, A);
```
https://godbolt.org/z/YErW4s5ed


The PR you linked probably doesn't help but fixing this bug here should be much simpler than implementing bit casts for bit fields.
